tests/unit/adapters/application-test.js

```diff
@@ -502,8 +698,10 @@ test('#fetch handles 4xx (Client Error) response status', function(assert) {
   });
   let promise = adapter.fetch('/posts', { method: 'POST', body: 'json string here' });
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  promise.catch(function(error) {
+  promise.catch((error) => {
     assert.ok(error.name, 'Client Error', '4xx response throws a custom error');
+    // This one doesn't really test anything, does it? It checks if the mocked response
+    // above uses the correct format (errors array).
```

Comment:
I'm not sure we're talking about the same thing/line here. I was referring to the line(s) below, #740 (and #741). All they do is check if `error` is an array (which it should be, but that is spec-compliancy on the (mock)server side). As in, you could "break" the errors response on line #730, which will break the tests, but hasn't proven anything meaningful about this lib/code itself. That's what I mean by "not testing anything really".

It can't hurt to do these checks though. I think this can be improved by doing another `deepEqual` and using an object in both the `resolve` on #730 and the `assert`s below. I'll update the PR.